todo/src/scripts/storageController.js

```diff
@@ -0,0 +1,88 @@
+class StorageController {
+    constructor(name) {
+        this.init();
+        this.data;
+    }
+
+    init() {
+        if (!localStorage.getItem('toDo')){
+            localStorage.setItem('toDo', JSON.stringify(
+                {
+                    'currentTasks': [],
+                    'completeTasks': [],
+                }
+            ));
+        }
+        this.data = JSON.parse(localStorage.getItem('toDo'));
+        this.removeEmptyItems();
+    }
+
+    removeEmptyItems() {
+        this.data['currentTasks'] = this.data['currentTasks'].filter((item) => {return item != null});
+        this.data['completeTasks'] = this.data['completeTasks'].filter((item) => {return item != null});    
+    }
+
+    updateStorage() {
+        localStorage.setItem('toDo', JSON.stringify(this.data));
+        this.data = JSON.parse(localStorage.getItem('toDo'));
+    }
+    
+    getItemsList() {
+        return this.data;
+    }
+
+    getCurentItemsList() {
+        return this.data['currentTasks'];
+    }
+
+    getCompleteItemsList() {
+        return this.data['completeTasks'];
+    }
+
+    addNewItem(title, task, priority, date) {
+        this.data.currentTasks.push(
+            {
+                'title': title,
+                'task': task,
+                /* 'date': date, */
+                'priority': priority,
+                'date': date
+            }
+        )
+        localStorage.setItem('toDo', JSON.stringify(this.data));
+    }
+
+    changeItemStatus(id) {
+        let currentTaskList = this.getCurentItemsList();
+        const task = currentTaskList[id];
+        this.getCompleteItemsList().push(task);
+        this.data.currentTasks.splice(id, 1, null);
+        this.updateStorage();
+        return task;
+    }
+
+    removeCurrentItem(id, updateData) {
+        if(updateData) {
+            this.data.currentTasks[id] = updateData;
+        } else {
+            this.data.currentTasks.splice(id, 1, null);
+        }
```

Comment:
use `filter` to remove item and you method violate SOLID, it should do 1 action remove according to your name, but you update and remove here.

Reply:
I changed to way of storing data and rewrite method.
```suggestion
changeItemStatus(id) {
        let currentTaskIndex = this.data.currentTasksId.indexOf(id);
        if (currentTaskIndex >= 0) {
            this.data.currentTasksId.splice(currentTaskIndex, 1);
            this.data.completeTasksId.push(id);
        } else {
            const uncompleteTaskIndex = this.data.completeTasksId.indexOf(id);
            this.data.completeTasksId.splice(uncompleteTaskIndex, 1);
            this.data.currentTasksId.push(id);
        }
        this.sortTasksId();
        this.updateStorage();
    }
```